app/mixins/controller/billing.js

```diff
@@ -0,0 +1,18 @@
+import Mixin from '@ember/object/mixin';
+import { reads } from '@ember/object/computed';
+import { inject as service } from '@ember/service';
+
+export default Mixin.create({
+  storage: service(),
+
+  account: reads('model.account'),
+  newSubscription: null,
+  queryParams: ['billingStep'],
+  billingStep: 1,
+
+  actions: {
+    updateBillingQueryParams(step) {
+      this.set('billingStep', step);
```

Comment:
Usually it's better to update queryParams by navigating to the new URL:
```js
this.transitionTo({ queryParams: { billingStep: step }});
```
Because the purpose of query params is to persist certain  portion of state to URL and then expand it after navigation, it's a bit more solid to have it one-way: navigate to route -> expand query parameter -> apply state -> state changed -> navigate to route -> expand... and so on. Although, currently router is not involved in the process of updating query parameter.

Reply:
You can also use it anywhere in the app, even on another routes: `router.transitionTo('routePath', { queryParams: { billingStep: step }})`